controllers/__init__.py

```diff
@@ -83,8 +82,7 @@ def update_status(self, id: str, status: str) -> Dict:
             return_document=ReturnDocument.AFTER
         )
         if not pipeline:
-            msg = f"Pipeline {id} not found."
-            raise NotFound(msg)
```

Comment:
Ideally on our controllers it shouldn't be raising web context exceptions, even though currently it's only used from a `@rest`, one day it can also be used from other points like event handlers, so we typically try keep our controllers either returning `Optional` for these cases or a custom exception. So, usually returning an `Optional` is the lightest one that we've been using.

Reply:
Got it.